data-prepper-plugins/grok-processor/src/main/resources/grok-patterns/patterns

```diff
@@ -0,0 +1,19 @@
+ALB_ACCESS_LOG_GENERAL_URI %{BASE_ALB_ACCESS_LOG_GENERAL_URI}\s\"(?:-|%{DATA:domain_name})\"\s\"(?:-|%{DATA:chosen_cert_arn})\"\s(?:-|%{INT:matched_rule_priority:int})\s%{TIMESTAMP_ISO8601:request_creation_time}\s\"(?:-|%{DATA:actions_executed})\"\s\"(?:-|%{DATA:redirect_url})\"\s\"(?:-|%{DATA:error_reason})\"\s\"(?:-|%{DATA:target_list})\"\s\"(?:-|%{DATA:target_status_code_list})\"\s\"(?:-|%{DATA:classification})\"\s\"(?:-|%{DATA:classification_reason})\"
+ALB_ACCESS_LOG %{BASE_ALB_ACCESS_LOG}\s\"(?:-|%{DATA:domain_name})\"\s\"(?:-|%{DATA:chosen_cert_arn})\"\s(?:-|%{INT:matched_rule_priority:int})\s%{TIMESTAMP_ISO8601:request_creation_time}\s\"(?:-|%{DATA:actions_executed})\"\s\"(?:-|%{DATA:redirect_url})\"\s\"(?:-|%{DATA:error_reason})\"\s\"(?:-|%{DATA:target_list})\"\s\"(?:-|%{DATA:target_status_code_list})\"\s\"(?:-|%{DATA:classification})\"\s\"(?:-|%{DATA:classification_reason})\"
+
+ELB_URIPATHPARAM %{URIPATH:path}(?:%{URIPARAM:params})?
+ELB_URI %{URIPROTO:proto}://(?:%{USER}(?::[^@]*)?@)?(?:%{URIHOST:urihost})?(?:%{ELB_URIPATHPARAM})?
+ELB_REQUEST_LINE (?:%{WORD:verb} %{ELB_URI:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})
+ELB_ACCESS_LOG %{TIMESTAMP_ISO8601:timestamp}\s%{NOTSPACE:elb}\s%{IP:clientip}:%{INT:clientport:int}\s(?:-|(?:%{IP:backendip}:?:%{INT:backendport:int}))\s(?:-|%{NUMBER:request_processing_time:float})\s(?:-|%{NUMBER:backend_processing_time:float})\s(?:-|%{NUMBER:response_processing_time:float})\s(?:-|%{INT:response:int})\s(?:-|%{INT:backend_response:int})\s(?:-|%{INT:received_bytes:int})\s(?:-|%{INT:bytes:int})\s\"%{ELB_REQUEST_LINE}\"
+
+S3_HTTP_REQUEST ((?:%{WORD:verb}\s%{NOTSPACE:request}\s(?:HTTP/%{NUMBER:httpversion}))?|%{DATA:rawrequest})
+S3_ACCESS_LOG %{WORD:owner}\s%{NOTSPACE:bucket}\s\[%{HTTPDATE:timestamp}\]\s%{IP:clientip}\s%{NOTSPACE:requester}\s%{NOTSPACE:request_id}\s%{NOTSPACE:operation}\s%{NOTSPACE:key}\s(?:-|"%{S3_HTTP_REQUEST}")\s(?:-|%{INT:response:int})\s(?:-|%{NOTSPACE:error_code})\s(?:-|%{INT:bytes_sent:int})\s(?:-|%{INT:object_size:int})\s(?:-|%{INT:request_time_ms:int})\s(?:-|%{INT:turnaround_time_ms:int})\s(?:%{QS:referrer})\s(?:-|"?%{QS:agent}"?)\s(?:-|%{NOTSPACE:version_id})
+
+CLOUDFRONT_ACCESS_LOG (?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY}\s%{TIME})\s%{NOTSPACE:x_edge_location}\s(?:-|%{NUMBER:sc_bytes:int})\s%{IPORHOST:clientip}\s%{WORD:cs_method}\s%{HOSTNAME:cs_host}\s%{NOTSPACE:cs_uri_stem}\s(?:-|%{NUMBER:sc_status:int})\s%{GREEDYDATA:referrer}\s%{GREEDYDATA:agent}\s%{GREEDYDATA:cs_uri_query}\s%{GREEDYDATA:cookies}\s%{WORD:x_edge_result_type}\s%{NOTSPACE:x_edge_request_id}\s%{HOSTNAME:x_host_header}\s%{URIPROTO:cs_protocol}\s(?:-|%{INT:cs_bytes:int})\s(?:-|%{GREEDYDATA:time_taken:float}\s%{GREEDYDATA:x_forwarded_for}\s%{GREEDYDATA:ssl_protocol}\s%{GREEDYDATA:ssl_cipher}\s%{GREEDYDATA:x_edge_response_result_type}
```

Comment:
) missing at the end :)

Reply:
@miklapko , Thank you for catching this and bringing it to our attention. Would you be interested in submitting a PR to fix it?

Reply:
@dlvenable Hi, sorry I'm kinda too overwhelmed with work atm, not sure when I can look into creating PRs for this. Maybe a few weeks later...

But also, I've found that this works for my ingestion pipeline (with newline codec) for Cloudfront -> OpenSearch
`(?<datetime>%{YEAR}-%{MONTHNUM}-%{MONTHDAY}\t%{TIME})\t%{NOTSPACE:x_edge_location}\t%{NUMBER:sc_bytes:int}\t%{IP:client_ip}\t%{WORD:cs_method}\t%{NOTSPACE:cs_host}\t%{NOTSPACE:cs_uri_stem}\t%{NUMBER:sc_status:int}\t%{NOTSPACE:cs_referer}\t%{NOTSPACE:cs_user_agent}\t%{NOTSPACE:cs_uri_query}\t%{NOTSPACE:cs_cookie}\t%{WORD:x_edge_result_type}\t%{NOTSPACE:x_edge_request_id}\t%{NOTSPACE:x_host_header}\t%{NOTSPACE:cs_protocol}\t%{NUMBER:cs_bytes:int}\t%{NUMBER:time_taken:float}\t%{NOTSPACE:x_forwarded_for}\t%{NOTSPACE:ssl_protocol}\t%{NOTSPACE:ssl_cipher}\t%{WORD:x_edge_response_result_type}\t%{NOTSPACE:cs_protocol_version}\t%{NOTSPACE:fle_status}\t%{NOTSPACE:fle_encrypted_fields}\t%{NUMBER:c_port:int}\t%{NUMBER:time_to_first_byte:float}\t%{NOTSPACE:x_edge_detailed_result_type}\t%{NOTSPACE:sc_content_type}\t%{NUMBER:sc_content_len:int}\t%{NOTSPACE:sc_range_start}\t%{NOTSPACE:sc_range_end}`
Probably needs further refinement but it's good enough in my use case it seems.